packages-internal/test-utils/src/createRenderer.tsx

```diff
@@ -301,7 +301,7 @@ function render(
   );
   const result: MuiRenderResult = {
     ...testingLibraryRenderResult,
-    user: userEvent.setup(),
+    user: userEvent.setup({ delay: null }),
```

Comment:
I've looked into failing tests on https://github.com/mui/mui-x/pull/14668 and I'm not sure we should move forward with this. 🤔 
Is it really such a big problem to keep with the existing `setTimeout` wrapping for each `userEvent` call? 🤔 
I wouldn't like to significantly degrade the DX (I wasn't able to fix all the failing tests without resetting `delay` to `0` on some of them) of writing tests for a negligible performance difference. 🙈 🤷

Reply:
I think we could let this PR rest until we've
* made all `act` async
* refactor fake timers to scope to single tests only
* migrated to vitest

and then re-evaluate how broken it is.

Reply:
It makes sense, let's go with it. 👍

Reply:
> Is it really such a big problem to keep with the existing setTimeout wrapping for each userEvent call? 

@LukasTy I think that it boils down to this: https://github.com/mui/material-ui/pull/43757#discussion_r1763177876. If moving from fireEvent to userEvent x2 the CI run time (I don't know if true), then IMHO, we shouldn't. Feedback time on PR is already painfully slow, it feels more important.

> I wasn't able to fix all the failing tests without resetting delay to 0 on some of them

Could you pinpoint to why? I mean Is there a technical reason why those setTimeout should be needed for tests to pass?

If not, it sounds like the right move is to have `{ delay: null }` in as many places as possible in the codebase as soon as possible to avoid regressions.

Reply:
> Could you pinpoint to why? I mean Is there a technical reason why those setTimeout should be needed for tests to pass?

The main problem is the `TransitionGroup` on Pickers tests.
There are errors thrown about mutating state, because of it.